stock_available_immediately/models/product_product.py

```diff
@@ -3,18 +3,20 @@
 # Copyright 2016 Sodexis
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import models
+from odoo import api, models
 
 
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
-    def _product_available(self, field_names=None, arg=False):
-        res = super(ProductProduct, self)._product_available(
-            field_names=field_names, arg=arg)
-        for prod_id in res:
-            res[prod_id]['immediately_usable_qty'] =\
-                res[prod_id]['immediately_usable_qty'] - \
-                res[prod_id]['incoming_qty']
-
+    @api.multi
+    def _compute_available_quantities_dict(self):
+        res = super(ProductProduct, self)._compute_available_quantities_dict()
+        for product in self:
+            res[product.id]['immediately_usable_qty'] -= \
+                product.incoming_qty
         return res
+
+    @api.depends('virtual_available', 'incoming_qty')
+    def _compute_available_quantities(self):
```

Comment:
What is the purpose of this method declaration? Are the depends changing or something?

Maybe add a docblock indicating the reasoning? Otherwise it's just a method calling the super.